src/cluster.c

```diff
@@ -2651,7 +2651,7 @@ void clusterProcessPingExtensions(clusterMsg *hdr, clusterLink *link) {
             clusterNode *n = clusterLookupNode(forgotten_node_ext->name, CLUSTER_NAMELEN);
             if (n && n != myself && !(nodeIsSlave(myself) && myself->slaveof == n)) {
                 sds id = sdsnewlen(forgotten_node_ext->name, CLUSTER_NAMELEN);
-                dictEntry *de = dictAddRaw(server.cluster->nodes_black_list, id, NULL);
+                dictEntry *de = dictAddOrFind(server.cluster->nodes_black_list, id);
```

Comment:
Do you plan on reverting this after you found the bug, otherwise the assert does nothing since it always returns something.

Reply:
I deleted the assert, but maybe we should not extend the expire timestamp when the dict entry already exists. That can lead to a situation where we prolong the blacklisted nodes indefinitely.